docs/global/modules/core/pages/howto/modelql.adoc

```diff
@@ -0,0 +1,190 @@
+val result: List<String?> = client.query { root ->
+    root.children("modules").property("name").toList()
+}
+--
+
+== Integration with LightModelClient
+
+When creating a `LightModelClient` you can optionally provide a `ModelQLClient` instance,
+which allows to invoke `.query { ... }` (see below) on a node returned by the `LightModelClient`.
+
+[source,kotlin]
+--
+val modelqlClient = ModelQLClient.builder().build()
+val client = LightModelClient.builder().modelQLClient(modelqlClient).build()
+val result: List<String?> = client.getRootNode()!!.query {
+    it.children("modules").property("name").toList()
+}
+--
+
+== Type safe ModelQL API
+
+You can use the `model-api-gen-gradle` plugin to generate type safe extensions from your meta-model.
+
+[source,kotlin]
+--
+val result: List<StaticMethodDeclaration> = client.query { root ->
+    root.children("classes").ofConcept(C_ClassConcept)
+        .member
+        .ofConcept(C_StaticMethodDeclaration)
+        .filter { it.visibility.instanceOf(C_PublicVisibility) }
+        .toList()
+}
+--
+
+== Run query on an INode
+
+If a query returns a node you can execute a new query starting from that node.
+
+[source,kotlin]
+--
+val cls: ClassConcept = client.query {
+    it.children("classes").ofConcept(C_ClassConcept).first()
+}
+val names = cls.query { it.member.ofConcept(C_StaticMethodDeclaration).name.toList() }
+--
+
+Or you can just use the INode API to access further data of that node.
+This is not recommended though, because each access sends a new query to the server.
+
+[source,kotlin]
+--
+val cls: ClassConcept = client.query {
+    it.children("classes").ofConcept(C_ClassConcept).first()
+}
+val className = cls.name
+--
+
+== Complex query results
+
+While returning a list of elements is simple
+the purpose of the query language is to reduce the number of request to a minimum.
+This requires to combine multiple values into more complex data structures.
+The `zip` operation provides a simple way of doing that:
+
+[source,kotlin]
+--
+val result: List<IZip3Output<Any, Int, String, List<String>>> = query { db ->
+    db.products.map {
+        val id = it.id
+        val title = it.title
+        val images = it.images.toList()
+        id.zip(title, images)
+    }.toList()
+}
+result.forEach { println("ID: ${it.first}, Title: ${it.second}, Images: ${it.third}") }
+--
+
+This is suitable for combining a small number of values,
+but because of the missing variable names it can be hard to read for a larger number of values
+or even multiple zip operations assembled into a hierarchical data structure.
+
+This can be solved by defining custom data classes and using the `mapLocal` operation:
+
+[source,kotlin]
+--
+data class MyProduct(val id: Int, val title: String, val images: List<MyImage>)
+data class MyImage(val url: String)
+
+val result: List<MyNonSerializableClass> = remoteProductDatabaseQuery { db ->
```

Comment:
Why is it important that the class is not serializable?

Including a definition of the class could also make the sample better readable.